crates/rpc-types-engine/src/attributes.rs

```diff
@@ -199,4 +225,70 @@ mod test {
         let extra_data = attributes.get_holocene_extra_data(BaseFeeParams::new(80, 60));
         assert_eq!(extra_data.unwrap(), Bytes::copy_from_slice(&[0, 0, 0, 0, 80, 0, 0, 0, 60]));
     }
+
+    #[test]
+    fn test_serde_roundtrip_attributes_pre_jovian() {
+        let attributes = OpPayloadAttributes {
+            payload_attributes: PayloadAttributes {
+                timestamp: 0x1337,
+                prev_randao: B256::ZERO,
+                suggested_fee_recipient: Address::ZERO,
+                withdrawals: Default::default(),
+                parent_beacon_block_root: Some(B256::ZERO),
+            },
+            transactions: Some(vec![b"hello".to_vec().into()]),
+            no_tx_pool: Some(true),
+            gas_limit: Some(42),
+            eip_1559_params: Some(b64!("0000dead0000beef")),
+            jovian_eip_1559_params: None,
+        };
+
+        let ser = serde_json::to_string(&attributes).unwrap();
+        let de: OpPayloadAttributes = serde_json::from_str(&ser).unwrap();
+
+        assert_eq!(attributes, de);
+    }
+
+    #[test]
+    fn test_serde_roundtrip_attributes_post_jovian() {
+        let attributes = OpPayloadAttributes {
+            payload_attributes: PayloadAttributes {
+                timestamp: 0x1337,
+                prev_randao: B256::ZERO,
+                suggested_fee_recipient: Address::ZERO,
+                withdrawals: Default::default(),
+                parent_beacon_block_root: Some(B256::ZERO),
+            },
+            transactions: Some(vec![b"hello".to_vec().into()]),
+            no_tx_pool: Some(true),
+            gas_limit: Some(42),
+            eip_1559_params: None,
+            jovian_eip_1559_params: Some(fixed_bytes!("000000080000000801")),
+        };
+
+        let ser = serde_json::to_string(&attributes).unwrap();
+        let de: OpPayloadAttributes = serde_json::from_str(&ser).unwrap();
+
+        assert_eq!(attributes, de);
+    }
+
+    #[test]
+    fn test_get_extra_data_post_jovian() {
+        let attributes = OpPayloadAttributes {
+            jovian_eip_1559_params: Some(fixed_bytes!("000000080000000801")),
+            ..Default::default()
+        };
+        let extra_data = attributes.get_jovian_extra_data(BaseFeeParams::new(80, 60));
```

Comment:
This test needs to be updated with `BaseFeeParams::new(80, 60, MIN_BASE_FEE_LOG2)` after the previous comment is addressed

Reply:
since `minBaseFeeLog2` will be a field in the payload then we don't